whynotearth/BrowTricks
# Feature: Renaming/Restructuring views and components

# Description

This PR does not address any particular issue, but involves some file renaming and restructuring for readability. In this PR, we:

* f429ede: Rename and restructure views and components to only have 1 level of nesting after 'src' folder. We follow how Volkswagen [components](https://github.com/whynotearth/Volkswagen-email-corporate-communications-CMS/tree/staging/src/components) and [views](https://github.com/whynotearth/Volkswagen-email-corporate-communications-CMS/tree/staging/src/views) are set up w.r.t. naming.

  Before: 
  
  ![image](https://user-images.githubusercontent.com/20192983/81490128-04383000-9233-11ea-9217-79f9f60e33f0.png)

  After: 
  
  ![image](https://user-images.githubusercontent.com/20192983/81490130-0ac6a780-9233-11ea-8ef0-edbf6dd7391e.png)

# Commits

* f429ede: Rename files as described above

* 5be72e6: Increase padding left of an element in 'MenuHeader', very minor

* 51b14bd: Move the `disable-scrollbars` styles into their own CSS file and import to 'app.css', as this will be used elsewhere besides only 'TextArea' input component (see below commit)

* 6ec1547: We make the 'LayoutFixedScrollable' component to use. It will be the basic go-to layout with a fixed header, content, and fixed footer. We begin using the component in the existing customer views. We remove the now-unnecessary sticky and z-index classes on 'BaseMenuHeader' and reduce the z-indices on 'BaseMenuDrawer' elements accordingly.

## Added commits after reviews:

* f02f9ca: Pull temporary background CSS for customer views into its own style file.

* a2b452a, 2b66644: Add the' vue100vh' package and use it in the 'LayoutFixedScrollable' component to achieve proper full height screen on mobile

* 0180317: Add FIXME notes for code that will be ripped out later

* 7a9b0d0: Use actual component names when importing

* fd6bd05: Use getters for reading state instead reading state vars directly

* 203bd84: Use `@apply` to simplify temporary background CSS for customer views

# How to Test

There are no visual nor functional changes in this PR. Visit the three views and observe that there are no visual nor functional changes to speak of.

1. [Customer View - Home](https://deploy-preview-47--browtricks.netlify.app/shop/test-tenant-slug)
2. [Customer View - Business Hours](https://deploy-preview-47--browtricks.netlify.app/shop/test-tenant-slug/business-hours)
3. [Customer View - Contact Form](https://deploy-preview-47--browtricks.netlify.app/shop/test-tenant-slug/contact)

(Links will take you to the pages in the deploy preview)
